### DISCUSSION TITLE
Restarting securityonion found many errors in the service

### DISCUSSION THREAD

**Discussion by shushu1234** [Security-Onion-Solutions/securityonion#2365] [Unsupported Versions]
My server accidentally lost power, and then I restarted securityonion, and found that many services have errors. How can I solve it?

[root@securityonion elasticsearch]# so-status 

Checking Docker status

    Docker -------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    

Checking container statuses

    so-aptcacherng ------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-cortex ----------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-curator ---------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-dockerregistry --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-elastalert ---------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-elasticsearch ------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------------ [ ERROR ]    
    so-fleet --------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-grafana ------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-idstools --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-influxdb --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-kibana ----------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-kratos ----------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------------ [ ERROR ]    
    so-mysql --------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-nginx ------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-playbook ------------------------------------------------------------------------------------------------------------------------------------------ [ ERROR ]    
    so-redis ------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-sensoroni ----------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-soc ----------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-soctopus --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-steno --------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-backend ----------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-coordinator ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-filestream -------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-frontend ---------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-gatekeeper ----------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-manager -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-suricata ---------------------------------------------------------------------------------------------------------------------------------------- [ PENDING ]    
    so-telegraf --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-thehive ---------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-thehive-es ---------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-wazuh --------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]

Thanks！

**Comment by stijnpieters** [comment]
If this issue persists, check if the docker containers are running using `docker ps`. If you cannot see the containers, something might have gone wrong with the docker images or salt.
First of all try applying `sudo salt '*' state.highstate`. If this does not fix it, try the following.
Try running `so-docker-refresh` this will refresh the docker images in the docker registry running on your manager node. Then try applying highstate again.

**Reply by rwwiv** [reply]
To add on to this, in previous releases `so-status` would show services in the `ERROR` state on boot until they started up. We improved the status reporting in 2.3.10 and this was something we specifically sought to improve. `so-status` will now show the status of services as `WAIT_START` on boot to make it clear the OS is in the process of starting up. We would suggest running `soup` to make sure you're on the latest version of Security Onion.

**Reply by shushu1234** [reply]
[root@securityonion ~]# docker ps -a
CONTAINER ID        IMAGE                                                           COMMAND                  CREATED             STATUS                          PORTS                                                                                                      NAMES
58c967bc2648        securityonion:5000/securityonion/so-suricata:2.3.10             "/usr/local/sbin/so-…"   32 minutes ago      Created                                                                                                                                    so-suricata
3b9c935c33f1        securityonion:5000/securityonion/so-mysql:2.3.10                "/entrypoint.sh mysq…"   16 hours ago        Created                         0.0.0.0:3306->3306/tcp, 33060/tcp                                                                          so-mysql
eeb6572625e8        securityonion:5000/securityonion/so-elastalert:2.3.10           "/opt/elastalert/run…"   19 hours ago        Exited (1) 13 minutes ago                                                                                                                  so-elastalert
f7f564b2221e        securityonion:5000/securityonion/so-wazuh:2.3.10                "/entrypoint.sh"         8 days ago          Exited (255) 40 hours ago       0.0.0.0:1514-1515->1514-1515/tcp, 0.0.0.0:55000->55000/tcp, 0.0.0.0:1514->1514/udp                         so-wazuh
0716804d58a0        securityonion:5000/securityonion/so-grafana:2.3.10              "/run.sh"                8 days ago          Exited (255) 40 hours ago       0.0.0.0:3000->3000/tcp                                                                                     so-grafana
50494a0a2669        securityonion:5000/securityonion/so-influxdb:2.3.10             "/entrypoint.sh infl…"   8 days ago          Up 37 minutes                   0.0.0.0:8086->8086/tcp                                                                                     so-influxdb
6fe5f0d94b7b        securityonion:5000/securityonion/so-telegraf:2.3.10             "/entrypoint.sh tele…"   8 days ago          Up 37 minutes                                                                                                                              so-telegraf
d3c78b122a5e        securityonion:5000/securityonion/so-logstash:2.3.10             "/usr/local/bin/dock…"   8 days ago          Exited (255) 40 hours ago       0.0.0.0:5044->5044/tcp, 0.0.0.0:5644->5644/tcp, 0.0.0.0:6050-6053->6050-6053/tcp, 0.0.0.0:9600->9600/tcp   so-logstash
0597a7022207        securityonion:5000/securityonion/so-strelka-filestream:2.3.10   "strelka-filestream"     8 days ago          Exited (255) 40 hours ago                                                                                                                  so-strelka-filestream
1a41c6f3d9fc        securityonion:5000/securityonion/so-strelka-manager:2.3.10      "strelka-manager"        8 days ago          Up 31 minutes                                                                                                                              so-strelka-manager
0550231b3af1        securityonion:5000/securityonion/so-strelka-backend:2.3.10      "strelka-backend"        8 days ago          Exited (0) About a minute ago                                                                                                              so-strelka-backend
e6af694955ec        securityonion:5000/securityonion/so-strelka-frontend:2.3.10     "strelka-frontend"       8 days ago          Exited (255) 40 hours ago       0.0.0.0:57314->57314/tcp                                                                                   so-strelka-frontend
d45267f43365        securityonion:5000/securityonion/so-redis:2.3.10                "redis-server --save…"   8 days ago          Up 32 minutes                   0.0.0.0:6381->6379/tcp                                                                                     so-strelka-gatekeeper
b5c4b9b5fad5        securityonion:5000/securityonion/so-redis:2.3.10                "redis-server --save…"   8 days ago          Up 32 minutes                   0.0.0.0:6380->6379/tcp                                                                                     so-strelka-coordinator
76ca295586d7        securityonion:5000/securityonion/so-thehive:2.3.10              "/opt/thehive/bin/so…"   8 days ago          Up 28 minutes                   0.0.0.0:9000->9000/tcp                                                                                     so-thehive
c4e12283a22a        securityonion:5000/securityonion/so-thehive-cortex:2.3.10       "/opt/cortex/bin/cor…"   8 days ago          Up 28 minutes                   0.0.0.0:9001->9001/tcp                                                                                     so-cortex
9fbe47c6d92e        securityonion:5000/securityonion/so-thehive-es:2.3.10           "/usr/local/bin/dock…"   8 days ago          Exited (255) 40 hours ago       9200/tcp, 0.0.0.0:9400->9400/tcp, 9300/tcp, 0.0.0.0:9500->9500/tcp                                         so-thehive-es
2b88bb112f5f        securityonion:5000/securityonion/so-filebeat:2.3.10             "/usr/local/bin/dock…"   8 days ago          Exited (255) 40 hours ago       0.0.0.0:514->514/tcp, 0.0.0.0:514->514/udp                                                                 so-filebeat
24fff414ef21        securityonion:5000/securityonion/so-nginx:2.3.10                "nginx -g 'daemon of…"   8 days ago          Up 37 minutes                   0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp, 0.0.0.0:8090->8090/tcp                                           so-nginx
99278e898a08        securityonion:5000/securityonion/so-redis:2.3.10                "redis-server /usr/l…"   8 days ago          Up 32 minutes                   0.0.0.0:6379->6379/tcp, 0.0.0.0:9696->9696/tcp                                                             so-redis
9cce92470b2f        securityonion:5000/securityonion/so-fleet:2.3.10                "/startfleet.sh"         8 days ago          Exited (255) 40 hours ago       0.0.0.0:8080->8080/tcp                                                                                     so-fleet
25f445871afd        securityonion:5000/securityonion/so-soctopus:2.3.10             "gunicorn -b 0.0.0.0…"   8 days ago          Up 28 minutes                   0.0.0.0:7000->7000/tcp                                                                                     so-soctopus
f29c29dac267        securityonion:5000/securityonion/so-playbook:2.3.10             "/docker-entrypoint.…"   8 days ago          Exited (1) 12 minutes ago                                                                                                                  so-playbook
4b6188baaf4a        securityonion:5000/securityonion/so-kibana:2.3.10               "/usr/local/bin/so-k…"   8 days ago          Up 32 minutes                   0.0.0.0:5601->5601/tcp                                                                                     so-kibana
ff9ebcf0cfb6        securityonion:5000/securityonion/so-kratos:2.3.10               "/start-kratos.sh"       8 days ago          Up 37 minutes                   0.0.0.0:4433-4434->4433-4434/tcp                                                                           so-kratos
d0c3bdaa8b9b        securityonion:5000/securityonion/so-soc:2.3.10                  "/opt/sensoroni/sens…"   8 days ago          Exited (255) 40 hours ago       0.0.0.0:9822->9822/tcp                                                                                     so-soc
41af73470306        securityonion:5000/securityonion/so-curator:2.3.10              "/entrypoint.sh"         8 days ago          Up 31 minutes                                                                                                                              so-curator
e2700c5aa839        securityonion:5000/securityonion/so-soc:2.3.10                  "/opt/sensoroni/sens…"   8 days ago          Exited (255) 40 hours ago                                                                                                                  so-sensoroni
e28234083cc6        securityonion:5000/securityonion/so-steno:2.3.10                "/usr/local/sbin/so-…"   8 days ago          Exited (255) 40 hours ago                                                                                                                  so-steno
e4f7e8f4b47d        securityonion:5000/securityonion/so-elasticsearch:2.3.10        "/tini -- /usr/local…"   8 days ago          Exited (255) 40 hours ago       0.0.0.0:9200->9200/tcp, 0.0.0.0:9300->9300/tcp                                                             so-elasticsearch
33f158e11d96        securityonion:5000/securityonion/so-idstools:2.3.10             "./entrypoint.sh"        8 days ago          Up 37 minutes                                                                                                                              so-idstools
57c3f014de63        securityonion:5000/securityonion/so-acng:2.3.10                 "/sbin/so-entrypoint…"   8 days ago          Up 37 minutes                   0.0.0.0:3142->3142/tcp                                                                                     so-aptcacherng
ec902cb8f940        ghcr.io/security-onion-solutions/registry:latest                "/entrypoint.sh /etc…"   8 days ago          Up 37 minutes                   0.0.0.0:5000->5000/tcp                                                                                     so-dockerregistry


[root@securityonion ~]# sudo salt '*' state.highstate
securityonion_standalone:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 49336 and was started at 2020, Dec 18 02:46:37.476634 with jid 20201218024637476634
ERROR: Minions returned with non-zero exit code


[root@securityonion ~]# so-status


Checking Docker status

    Docker -------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    

Checking container statuses

    so-aptcacherng ------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-cortex ----------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-curator ---------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-dockerregistry --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-elastalert ---------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-elasticsearch ------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------------ [ ERROR ]    
    so-fleet --------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-grafana ------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-idstools --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-influxdb --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-kibana ----------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-kratos ----------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------------ [ ERROR ]    
    so-mysql ------------------------------------------------------------------------------------------------------------------------------------------- [ PENDING ]    
    so-nginx ------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-playbook ------------------------------------------------------------------------------------------------------------------------------------------ [ ERROR ]    
    so-redis ------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-sensoroni ----------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-soc ----------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-soctopus --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-steno --------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-backend ----------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-coordinator ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-filestream -------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-frontend ---------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-strelka-gatekeeper ----------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-manager -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-suricata ---------------------------------------------------------------------------------------------------------------------------------------- [ PENDING ]    
    so-telegraf --------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-thehive ---------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-thehive-es ---------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]    
    so-wazuh --------------------------------------------------------------------------------------------------------------------------------------------- [ ERROR ] 


I want to know what should I do next. Thanks!
